### PR TITLE
diff() feature with tests

### PR DESCRIPTION
**Pull request by jsntsay**
diff() feature takes an operator and generates a diff with the other operator. Useful when modifying an existing pipeline through customize_schema or the (upcoming) replace() feature and examining what changed. By default, displays Markdown diff for notebooks. 

Tests were added to tests/test_json_pretty_viz.py

### CODE REVIEW

File: lale/operators.py [hunk 1]
```diff
@@ -452,6 +453,63 @@ def pretty_print(
             markdown = IPython.display.Markdown(f"```python\n{result}\n```")
             return IPython.display.display(markdown)
 
+    def diff(
+        self,
+        other: Union[Any, "Operator"],
+        show_imports: bool = True,
+        customize_schema: bool = True,
+        ipython_display: bool = True,
```
Comment (hirzel) [inline lale/operators.py:461]: I would default `ipython_display` to False just like in `pretty_print`.

File: lale/operators.py [hunk 2]
```diff
@@ -452,6 +453,63 @@ def pretty_print(
             markdown = IPython.display.Markdown(f"```python\n{result}\n```")
             return IPython.display.display(markdown)
 
+    def diff(
+        self,
+        other: Union[Any, "Operator"],
```
Comment (hirzel) [inline lale/operators.py:458]: Why is this a `Union[Any, "Operator"]`? I would have expected just `"Operator"`.

File: lale/operators.py [hunk 3]
```diff
@@ -452,6 +453,63 @@ def pretty_print(
...
+            Operator to diff against
+
+        show_imports : bool, default True
+            Whether to include import statements in the pretty-printed code.
+
+        customize_schema : bool, default False
+            If True, then individual operators whose schema differs from the lale.lib version of the operator will be printed with calls to `customize_schema` that reproduce this difference.
+
+        ipython_display : bool, default True
+            If True, proactively ask Jupyter to render the graph.
```
Comment (hirzel) [inline lale/operators.py:477]: This looks like it was copy-and-pasted from the documentation of the `visualize` method, but here, you are not rendering a graph, but rather, using syntax highlighting.

File: lale/operators.py [hunk 4]
```diff
@@ -452,6 +453,63 @@ def pretty_print(
...
+            show_imports=show_imports,
+            ipython_display=False,
+        )
+        other_lines = other_str.splitlines()  # type: ignore
+
+        differ = difflib.Differ()
+        compare = differ.compare(self_lines, other_lines)
+
+        compare_str = "\n".join(compare)
+        if ipython_display is False:
```
Comment (hirzel) [inline lale/operators.py:505]: Since `ipython_display` is a boolean, you could simply test `if not ipython_display`.

**Review by hirzel — CHANGES_REQUESTED** [review]
Thanks a lot for your PR! It looks great overall. I especially appreciate the tests and documentation. I left a few minor inline comments.

File: lale/operators.py (hunk 1 above)
Comment (jsntsay) [reply lale/operators.py]: By the same token, I'm assuming `customize_schema` should default to False too then.

**Review by hirzel — APPROVED** [review]
Thanks for implementing the review feedback. Your PR looks great now, I have no further comments.

**Pull request merged**